pkg/logs/input/docker/tailer.go

```diff
@@ -36,53 +36,57 @@ const tagsUpdatePeriod = 10 * time.Second
 // Logs from stdout and stderr are multiplexed into a single channel and needs to be demultiplexed later one.
 // To multiplex logs, docker adds a header to all logs with format '[SEV][TS] [MSG]'.
 type Tailer struct {
-	ContainerID   string
-	outputChan    chan *message.Message
-	decoder       *decoder.Decoder
-	reader        io.ReadCloser
-	cli           *client.Client
-	source        *config.LogSource
-	containerTags []string
+	container  *Container
+	outputChan chan *message.Message
+	decoder    *decoder.Decoder
+	reader     io.ReadCloser
+	cli        *client.Client
+	source     *config.LogSource
+	// containersSource is the source that uses short image name as source and service
+	// tag if applicable (i.e. no k8s annotation nor docker label)
+	containerSource *config.LogSource
```

Comment:
why do we need both `source` and `containerSource` ? The tailer should only take on source and be agnostic from any upstream business logic